hedera-deduplication-bigquery/src/main/java/com/hedera/dedupe/IncrementalDeduplication.java

```diff
@@ -0,0 +1,110 @@
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.hedera.dedupe.query.GetLatestDedupeRowTemplateQuery;
+import com.hedera.dedupe.query.UpdateDedupeColumnTemplateQuery;
+
+@Component
+public class IncrementalDeduplication extends AbstractDeduplication {
+
+    private final AtomicLong delayGauge = new AtomicLong(0L);
+    private final UpdateDedupeColumnTemplateQuery updateDedupeColumnTemplateQuery;
+    private final GetLatestDedupeRowTemplateQuery getLatestDedupeRowTemplateQuery;
+    private final long incrementalProbeInteraval;
+
+    public IncrementalDeduplication(DedupeProperties properties, BigQuery bigQuery, MeterRegistry meterRegistry) {
+        super(DedupeType.INCREMENTAL, properties, bigQuery, meterRegistry);
+        updateDedupeColumnTemplateQuery = new UpdateDedupeColumnTemplateQuery(properties.getProjectId(),
+                properties.getTransactionsTableFullName(), bigQuery, meterRegistry);
+        getLatestDedupeRowTemplateQuery = new GetLatestDedupeRowTemplateQuery(properties.getProjectId(),
+                properties.getTransactionsTableFullName(), bigQuery, meterRegistry);
+        incrementalProbeInteraval = properties.getIncrementalProbeInterval();
+        Gauge.builder("dedupe.incremental.delay", delayGauge, AtomicLong::get)
+                .description("Delay in deduplication (now - startTimestamp)")
+                .baseUnit("sec")
+                .register(meterRegistry);
+    }
+
+    @Scheduled(fixedRateString = "${hedera.dedupe.incrementalFixedRate:300000}") // default: 5 min
+    public void run() {
+        runDedupe();
+    }
+
+    @Override
+    TimestampWindow getTimestampWindow(Map<String, FieldValue> state) throws InterruptedException {
+        long startTimestamp = 0;
+        if (state.containsKey(INCREMENTAL_LATEST_END_TIMESTAMP)) {
+            // no +1 since since we filter on non-unique column
+            startTimestamp = state.get(INCREMENTAL_LATEST_END_TIMESTAMP).getLongValue();
+        }
+        delayGauge.set(Duration.between(Instant.ofEpochSecond(0L, startTimestamp), Instant.now()).getSeconds());
+
+        long endTimestmap = probeEndTimestamp(startTimestamp);
+        return new TimestampWindow(startTimestamp, endTimestmap);
+    }
+
+    @Override
+    void saveState(TimestampWindow timestampWindow) throws InterruptedException {
+        setState.set(INCREMENTAL_LATEST_END_TIMESTAMP, String.valueOf(timestampWindow.getEndTimestamp()));
+    }
+
+    private long probeEndTimestamp(long startTimestamp) throws InterruptedException {
+        long endTimestamp = startTimestamp;
+        long intervalInSec = incrementalProbeInteraval;
+        while (true) {
```

Comment:
Might be a good idea to add iterations limit here to prevent infinite loop. An infinite loop is possible if we forgot to handle some edge case.